### DISCUSSION TITLE
[Auto3dseg] Transfomrations to get predictions on unseen data?

### DISCUSSION THREAD

**Discussion by amalmsr** [Project-MONAI/MONAI#7539] [Q&A]
I trained a SegRestNet model with the minimal input Auto3DSeg command, i didn't have any additional user-specific inputs. I used 
''' python -m monai. apps. auto3dseg AutoRunner run --input=./input. yaml --algos =segresnet '''

Now i want to perform inference on unseen data, as far as i learned, i need to use the data in the saved pytorch model and the used algorithm for SegResNet which is already implemented by monai.  But I want to know, if there any other transfomations for SegResNet, that are not mentioned in the 'algorith_generation.md' in ''' Project-MONAI/tutorials/tree/main/auto3dseg/docs/algorithm_generation.md'''?

Also any other easier suggestions for inference are very welcome! my problem with ensemblrrunner class is I am unsure what to put in kwargs, so I thought I'll try to rebuild the model manually with the available monai docs.

**Comment by KumoLiu** [comment]
Hi @spes3, thanks for your interest here.
Here is a tutorial about how to customize your ensemble pipeline by defining a new ensemble class.
https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/notebooks/ensemble_byoc.ipynb

Hope it helps, thanks.

**Reply by amalmsr** [reply]
Thank you for taking the time to answer, so here,  which part gets the transformations? My question is mainly about the transformations.

**Reply by KumoLiu** [reply]
You can find all the configs which contains transforms under each algo's folder such as "segresnet_0" and "dints_0".

**Reply by amalmsr** [reply]
SegResNet only has hyperparameters under config, and in segmenter.py I found custom transofmations as well as the default ones. Im going to try again with that and ask again if i get any issues. Thank you very much for your time, help and the quick answers.

**Comment by amalmsr** [comment]
in case anyone got stuck like me, these transforms did the job for me, they should be found in segmenter.py
transforms_infer = Compose([
    LoadImage(image_only=True),
    EnsureChannelFirst(),
    ScaleIntensityRange(a_min=(from hyper_parameters.yaml.), a_max=(from hyper_parameters.yaml.), b_min=0.0, b_max=1.0, clip=True),
    Spacing(pixdim=[1.0, 1.0, 1.0], mode='bilinear', align_corners=True),
    ToTensor(),
])
this is what i used, then SlidingWindowInfererAdapt with the roi_size from hyper_parameters.yaml. then the post_processing to save the output
